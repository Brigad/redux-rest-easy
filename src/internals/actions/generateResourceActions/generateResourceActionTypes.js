import generateActionType from '../../utils/generateActionType';

const generateResourceActionTypes = resourceName => ({
  INVALIDATE_RESOURCE: generateActionType(
    resourceName,
    '@invalidate',
    'INVALIDATE_RESOURCE',
  ),
  INVALIDATE_ID: generateActionType(
    resourceName,
    '@invalidate',
    'INVALIDATE_ID',
  ),
  RESET_RESOURCE: generateActionType(resourceName, '@reset', 'RESET_RESOURCE'),
});

export default generateResourceActionTypes;
