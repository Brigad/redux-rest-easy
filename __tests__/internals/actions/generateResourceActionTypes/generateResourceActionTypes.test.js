import generateResourceActionTypes from '../../../../src/internals/actions/generateResourceActions/generateResourceActionTypes';

describe('generateResourceActionTypes', () => {
  test('only path', () => {
    expect(generateResourceActionTypes('fruits')).toMatchSnapshot();
  });
});
