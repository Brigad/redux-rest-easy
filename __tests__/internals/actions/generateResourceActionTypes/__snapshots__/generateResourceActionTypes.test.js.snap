// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateResourceActionTypes only path 1`] = `
Object {
  "INVALIDATE_ID": "@@rest-easy/fruits/@invalidate/INVALIDATE_ID",
  "INVALIDATE_RESOURCE": "@@rest-easy/fruits/@invalidate/INVALIDATE_RESOURCE",
  "RESET_RESOURCE": "@@rest-easy/fruits/@reset/RESET_RESOURCE",
}
`;
